python/cudf/cudf/core/column/decimal.py

```diff
@@ -364,18 +364,51 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     else:
```

Comment:
I see a comment above,
> This should at some point be hooked up to libcudf's binary_operation_fixed_point_scale

Do we only support add/sub/mul/div operations right now in Python because of limitations in this function? I know that other operations are implemented, so piping that through might be a significant improvement.

Reply:
> Do we only support add/sub/mul/div operations right now in Python because of limitations in this function?


Not just `binary_operation_fixed_point_scale` but I think support for other binop's are not supported from libcudf side. 

Looking into `binary_operation_fixed_point_scale`, it seems the formula for `DIV` is wrong? I could be wrong here but don't match what is specified here: https://docs.microsoft.com/en-us/sql/t-sql/data-types/precision-scale-and-length-transact-sql

Though libcudf doesn't take `precision` as input the python side will need calculation so probably better to have those two computations in a single place rather than having to have to look at two places.

Reply:
Support for other operators exists, e.g. `MOD` / `PMOD` / `PYMOD`: https://github.com/rapidsai/cudf/pull/10179.

I'm fine with keeping both precision/scale calculations together here. I just wanted to make a note to ask, since I saw the comment above.

There may or may not be issues with the scale/precision calculations. I think the page you referenced has different conventions than libcudf. In my understanding:
- libcudf's scale represents powers of the radix (base 10 or base 2)
- libcudf's precision (32, 64, 128) represents bits (powers of two) used to store the integral part
Neither value appears to correspond to the linked SQL docs. That page appears to always use powers of 10 for both scale and precision. Also the definition of scale is the negative of libcudf's definition. It does not surprise me that these different conventions would result in different expressions. I spent an hour looking into this but I have no idea how to make the two definitions mathematically correspond.

Working through an example calculation here, for the SQL docs:
```python
e1 = 4.096
p1 = 4
s1 = 3
e2 = 3.2
p2 = 2
s2 = 1
s = max(6, s1 + p2 + 1)
p = p1 - s1 + s2 + s
print(f"{e1/e2=}")  # e1/e2=1.28
print(f"{p=}, {s=}")  # p=8, s=6
```

I was confused and gave up at this point -- how could `1.28` have `p=8, s=6`?

Reply:
Thanks @bdice, I think @codereport would have a better understanding on this than me. But I'm merging these changes for now and we can have a follow-up PR if changes need to be done.